source/uwp/AdaptiveCardsObjectModel/lib/AdaptiveTextElement.h

```diff
@@ -17,13 +17,38 @@ namespace winrt::AdaptiveCards::ObjectModel::Uwp::implementation
     protected:
         template<typename T> void InitializeTextElement(const std::shared_ptr<T>& sharedModel)
         {
-            IsSubtle = sharedModel->GetIsSubtle();
-            FontType = opt_cast<Uwp::FontType>(sharedModel->GetFontType());
-            Size = opt_cast<Uwp::TextSize>(sharedModel->GetTextSize());
-            Weight = opt_cast<Uwp::TextWeight>(sharedModel->GetTextWeight());
-            Color = opt_cast<Uwp::ForegroundColor>(sharedModel->GetTextColor());
             Text = UTF8ToHString(sharedModel->GetText());
             Language = UTF8ToHString(sharedModel->GetLanguage());
+
+            if (sharedModel->GetIsSubtle().has_value())
```

Comment:
Huh, really curious about what I messed up with the `operator=` on `property_opt` here. Was it a compile error or runtime?

Reply:
It was a runtime error, the values were just not being set. At the time i hadn't realized that property_opt was your creation so i didn't look too hard at the code there and just updated the caller side. I went back and did what i should have don't in the first place and fixed the property_opt set method. Let me know if my change looks right to you.

Reply:
Huh! Thank you, I'd thought `std::optional<T>::operator bool()` ([see here](https://en.cppreference.com/w/cpp/utility/optional/operator_bool)) would have done this.  I'll have to dig a little more for my own edification. Your change looks great.